src/sagemaker/cli/compatibility/v2/modifiers/image_uris.py

```diff
@@ -0,0 +1,161 @@
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Classes to modify image uri retrieve methods for Python SDK v2.0 and later."""
+from __future__ import absolute_import
+
+import ast
+import logging
+import pasta
+
+from sagemaker.cli.compatibility.v2.modifiers import matching
+from sagemaker.cli.compatibility.v2.modifiers.modifier import Modifier
+
+GET_IMAGE_URI_NAME = "get_image_uri"
+GET_IMAGE_URI_NAMESPACES = (
+    "sagemaker",
+    "sagemaker.amazon_estimator",
+    "sagemaker.amazon.amazon_estimator",
+    "amazon_estimator",
+    "amazon.amazon_estimator",
+)
+
+ALGORITHM_NAME_FROM_REPO = {
+    "blazingtext": "blazingtext",
+    "factorization-machine": "factorization-machines",
+    "forecasting-deepar": "forecasting-deepar",
+    "image-classification": "image-classification",
+    "image-classification-neo": "image-classification-neo",
+    "ipinsights": "ipinsights",
+    "kmeans": "kmeans",
+    "knn": "knn",
+    "lda": "lda",
+    "linear-learner": "linear-learner",
+    "ntm": "ntm",
+    "object2vec": "object2vec",
+    "object-detection": "object-detection",
+    "pca": "pca",
+    "randomcutforest": "randomcutforest",
+    "semantic-segmentation": "semantic-segmentation",
+    "seq2seq": "seq2seq",
+    "sagemaker-xgboost": "xgboost",
+    "xgboost-neo": "xgboost-neo",
+}
+
+logger = logging.getLogger("sagemaker")
+
+
+class ImageURIRetrieveRefactor(Modifier):
+    """A class to refactor *get_image_uri() method."""
+
+    def node_should_be_modified(self, node):
+        """Checks if the ``ast.Call`` node calls a function of interest.
+
+        This looks for the following calls:
+
+        - ``sagemaker.get_image_uri``
+        - ``sagemaker.amazon_estimator.get_image_uri``
+        - ``get_image_uri``
+
+        Args:
+            node (ast.Call): a node that represents a function call. For more,
+                see https://docs.python.org/3/library/ast.html#abstract-grammar.
+
+        Returns:
+            bool: If the ``ast.Call`` instantiates a class of interest.
+        """
+        return matching.matches_name_or_namespaces(
+            node, GET_IMAGE_URI_NAME, GET_IMAGE_URI_NAMESPACES
+        )
+
+    def modify_node(self, node):
+        """Modifies the ``ast.Call`` node to call ``image_uris.retrieve`` instead.
+        And switch the first two parameters from (region, repo) to (framework, region)
+
+        Args:
+            node (ast.Call): a node that represents a *image_uris.retrieve call.
+        """
+        original_args = [None] * 3
+        for kw in node.keywords:
+            if kw.arg == "repo_name":
+                arg = kw.value.s
+                modified_arg = ALGORITHM_NAME_FROM_REPO[arg]
```

Comment:
I don't think you need the dictionary at all - even for XGBoost as a framework, `get_image_uri` accepted just "xgboost": https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/amazon/amazon_estimator.py#L660

Reply:
updated!